tests/serverpod_test_server/test_integration/security_context_test.dart

```
@@ -0,0 +1,292 @@
+        var client = HttpClient();
+        client.badCertificateCallback = (cert, host, port) => true;
+
+        HttpClientRequest request = await client.getUrl(
+          Uri.parse('https://localhost:8080'),
+        );
+        var response = await request.close();
+
+        expect(response.statusCode, 200);
+      });
+
+      test('then server cannot be accessed over http', () async {
+        var client = HttpClient();
+        client.badCertificateCallback = (cert, host, port) => true;
+
+        HttpClientRequest request = await client.getUrl(
+          Uri.parse('http://localhost:8080'),
+        );
+
+        expect(
+          () async => await request.close(),
+          //Since the server simply drops HTTP connections instead of negotiating SSL,
+          //HttpException is the correct error and not HandshakeException.
+          throwsA(isA<HttpException>()),
+        );
+      });
+    });
+
+    group('without a security context', () {
+      late Serverpod serverpod;
+
+      setUpAll(() async {
+        serverpod = IntegrationTestServer.create();
+
+        await serverpod.start();
+      });
+
+      tearDownAll(() async {
+        await serverpod.shutdown(exitProcess: false);
+      });
+
+      test('then server is running over http', () {
+        expect(serverpod.server.securityContext, isNull);
+      });
+
+      test('then server can be accessed over http', () async {
+        var client = HttpClient();
+
+        HttpClientRequest request = await client.getUrl(
+          Uri.parse('http://localhost:8080'),
+        );
+
+        var response = await request.close();
+
+        expect(response.statusCode, 200);
+      });
+
+      test('then server cannot be accessed over https', () async {
+        var client = HttpClient();
+
+        expect(
+          () async => await await client.getUrl(
+            Uri.parse('https://localhost:8080'),
+          ),
+          throwsA(isA<HandshakeException>()),
+        );
+      });
```

Comment: This test is basically the exact same as above but we are just not ignoring the error. Modifying the above test with the real cert this test makes more sense.

---

Which test above you are referring here?

---

disregard this